src/ralph/models/xapi/page_viewed.py

```diff
@@ -0,0 +1,119 @@
+"""PageViewed xAPI event definitions"""
+
+from datetime import datetime
+from ipaddress import IPv4Address
+from pathlib import Path
+from typing import Literal, Optional
+
+from pydantic import AnyUrl, Field, PrivateAttr, constr, validator
+
+from ..base import BaseModelWithConfig
+from .constants import (
+    EN,
+    PAGE,
+    VERSION,
+    VIEWED,
+    XAPI_ACTIVITY_PAGE,
+    XAPI_EXTENSION_ACCEPT_LANGUAGE,
+    XAPI_EXTENSION_AGENT,
+    XAPI_EXTENSION_COURSE_ID,
+    XAPI_EXTENSION_COURSE_USER_TAGS,
+    XAPI_EXTENSION_HOST,
+    XAPI_EXTENSION_IP,
+    XAPI_EXTENSION_ORG_ID,
+    XAPI_EXTENSION_PATH,
+    XAPI_EXTENSION_REQUEST,
+    XAPI_VERB_VIEWED,
+)
+
+
+class ActorAccountField(BaseModelWithConfig):
+    """Represents the `actor.account` xAPI field."""
+
+    name: constr(min_length=1) = "student"
+    homePage: constr(min_length=1)
+
+    @validator("name", pre=True, always=True)
+    def set_name(cls, name):  # pylint:disable=no-self-argument,no-self-use
+        """Sets the name to student when it's None or empty."""
+
+        return name or "student"
```

Comment:
If I understood it right, this would be conversion and validation at the same time ?
We pass the edx `context.user_id` to the pydantic `name` field without change. Before checking for `constr(min_length=1)`  pydantic runs the validator `set_name` which checks/prepares the `name` field.

Reply:
Is this validator useful? You can assign the default value to student directly, isnt'it what you are already doing? 
Or maybe another possibility would be to assign the type str and as you only have two attributes of str type you could use the config class.
The more I get to use pydantic, and the more I think validator are not always a good solution

```
class ActorAccountField(BaseModelWithConfig):
    """Represents the `actor.account` xAPI field."""

    name: str = "student"
    homePage: str

        class Config:
            min_anystr_length = 1
```

Reply:
I thing this is a confusing part of doing validation and conversion at the same time.
The validator actually does the "transfomation" part.
If we omit for example the homePage attribute: 
For `ActorAccountField(name="")`, `ActorAccountField(name=None)`, `ActorAccountField()`, `ActorAccountField(name={})`, etc. we would get `ActorAccountField(name="student")` and not raise exceptions.